fe-traveltrove/src/features/staff/components/CitiesTab.tsx

```diff
@@ -1,218 +1,282 @@
-                                    value={formData.name}
-                                    onChange={(e) => setFormData({ ...formData, name: e.target.value })}
-                                />
-                            </Form.Group>
-                            <Form.Group className="mb-3">
-                                <Form.Label>Country</Form.Label>
-                                <Form.Select
-                                    value={formData.countryId}
-                                    onChange={(e) => setFormData({ ...formData, countryId: e.target.value })}
-                                >
-                                    <option value="">Select a country</option>
-                                    {countries.map((country) => (
-                                        <option key={country.countryId} value={country.countryId}>
-                                            {country.name}
-                                        </option>
-                                    ))}
-                                </Form.Select>
-                            </Form.Group>
-                        </Form>
-                    )}
-                </Modal.Body>
-                <Modal.Footer>
-                    <Button variant="secondary" onClick={() => setShowModal(false)}>
-                        Cancel
-                    </Button>
-                    <Button
-                        variant={modalType === "delete" ? "danger" : "primary"}
-                        onClick={modalType === "delete" ? handleDelete : handleSave}
-                    >
-                        {modalType === "delete" ? "Confirm" : "Save"}
-                    </Button>
-                </Modal.Footer>
-            </Modal>
-        </div>
-    );
+      {/* Modals for Create, Update, and Delete */}
+      <Modal show={showModal} onHide={() => setShowModal(false)}>
+        <Modal.Header closeButton>
+          <Modal.Title>
+            {modalType === "create"
+              ? "Create City"
+              : modalType === "update"
+              ? "Edit City"
+              : "Delete City"}
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          {modalType === "delete" ? (
+            <p>Are you sure you want to delete this city?</p>
+          ) : (
+            <Form>
+              <Form.Group className="mb-3">
+                <Form.Label>City Name</Form.Label>
+                <Form.Control
+                  required
+                  type="text"
+                  value={formData.name}
+                  onChange={(e) => {
+                    setFormData({ ...formData, name: e.target.value });
+                    setNameError(false); // error is removed when the user starts typing
+                  }}
+                  isInvalid={nameError}
+                />
+                <div className="invalid-feedback">City name is required.</div>
+              </Form.Group>
```

Comment:
Well done! Implementing a fail safe warning that the fields must be filled before posting a new entity and editing an entity.